### DISCUSSION TITLE
OAuth 2 support

### DISCUSSION THREAD

**Discussion by Billos** [the-djmaze/snappymail#1677] [General]
Hello,
Is there any plan on supporting OAuth2 for account provisionning ?

**Comment by the-djmaze** [comment]
Partially with the login-gmail and login-0365 extensions, and also Nextcloud OIDC,
If you need something else feel free to implement.

**Reply by githubbbie** [reply]
Could you use generic login-gmail, login-0365 or Nexcloud OIDC for general OAuth2 support?

With the ability of increasingly amazing compution automation doing brute force password attacks, my thinking is imap username/passwords need some 2auth or more secure system moving forward in history.

For me, and maybe others, as you also have linked to docker-mailserver in a few places, I would prefer some better ability to secure my imap accounts (currently using username/password alone) on my docker-mailserver along the lines of OAuth2 implemented there that works then with snappymail with OAuth2 or similar enhancement.

Any ideas here? Maybe there is already a solution for better security between docker-mailserver and snappymail.

THANK YOU. Snappymail is amazing.

**Reply by the-djmaze** [reply]
You could use the TOTP extension.
The only benefit of this approach is prevention of login attacks where something tries to detect your password.

**Reply by githubbbie** [reply]
Yes @the-djmaze the TOTP extension for Snappymail helps improve security for Snappymail, but I'm looking deeper to the actual IMAP server also requriing something similar. Does this exist?

**Reply by the-djmaze** [reply]
> Does this exist?

Read https://documentation.open-xchange.com/7.10.4/middleware/mail/dovecot/oauth_2.0_with_postfix_and_dovecot.html

**Comment by githubbbie** [comment]
Looks like Docker Mail Server recommends Authentik to handle Oauth2

https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/auth-oauth2

Worting on now, once that is going, then will work on integrating with SnappyMail, which is amazing!

**Reply by githubbbie** [reply]
Wow, filed a couple of issues in testing gmail: #1779 #1778 

Got authentik working with docker, and then next step will be to try to refashion oauth2 plugins to support authentik.

**Comment by tilllt** [comment]
Hey, i am interested as well in OAuth2, especially since we we are using Authentik anyways. I can see there is a OAuth2 Plugin but i cannot find any documentation on it or the current state this plugin is in:

https://github.com/the-djmaze/snappymail/tree/master/plugins/login-oauth2

did anyone succeed with "generic" OAuth2 login per Authentik / Authelia et al?
